src/mail/export/Exporter.ts

```diff
@@ -65,16 +65,36 @@ export async function exportMails(
 	entityClient: EntityClient,
 	fileController: FileController,
 	operationId?: OperationId,
+	signal?: AbortSignal,
 ): Promise<void> {
 	// Considering that the effort for generating the bundle is higher
 	// than generating the files, we need to consider it twice, so the
 	// total effort would be (mailsToBundle * 2) + filesToGenerate
 	const totalMails = mails.length * 3
 	let doneMails = 0
+	let cancel = false
+
+	const onAbort = () => {
+		cancel = true
+	}
+
+	//We use once here to let the AbortController remove the
+	//listener after dispatch the abort function
+	signal?.addEventListener("abort", onAbort, { once: true })
```

Comment:
As a general note: please don't throw something that doesn't extend `Error`.
We also have have `CancelledError` already.

I would reorganize a bit where this error is caught though. I would catch it somewhere inside this function and return  early instead. Otherwise it's a bit far from where it is thrown and it's hard to follow.

With that you can also do `removeEventListener` inside `finally` block